src/utils/memoize.ts

```diff
@@ -15,18 +15,18 @@ export const memoize = <T>(
 }
 
 export function filememoizeAsync<F extends HasParent, T>(
-  fn: (file: F) => Promise<T>,
-): WithCache<(file: F) => Promise<T>> {
+  fn: (file: F, ...args: any[]) => Promise<T>,
+): WithCache<(file: F, ...args: any[]) => Promise<T>> {
   const cache = new Map<string, Map<F, T>>()
-  const cached = async function (this: any, file: F): Promise<T> {
+  const cached = async function (this: any, file: F, ...args: any[]): Promise<T> {
     let subcache = cache.get(file.parent.path)
     if (!subcache) {
       subcache = new Map()
       cache.set(file.parent.path, subcache)
     }
     let val = subcache.get(file)
     if (!val) {
-      val = await fn.call(this, file)
+      val = await fn.call(this, file, ...args)
       subcache.set(file, val)
     }
     return val
```

Comment:
It looks like in JS there's no equivalent to tuples, so we would need to key on a string.

```diff
-  const cache = new Map<string, Map<F, T>>()
+  const cache = new Map<string, Map<string, T>>()
   const cached = async function (this: any, file: F, ...args: any[]): Promise<T> {
     let subcache = cache.get(file.parent.path)
     if (!subcache) {
       subcache = new Map()
       cache.set(file.parent.path, subcache)
     }
-    let val = subcache.get(file)
+    const key = `${file.path}:${args.join(',')}`
+    let val = subcache.get(key)
     if (!val) {
       val = await fn.call(this, file, ...args)
-      subcache.set(file, val)
+      subcache.set(key, val)
     }
```

Not sure it's worth it.

Reply:
Can kind of kludge tuples with this:
```
let worstTuple: [string, number];
```

Forces first element to be a string and second to be a number.

Reply:
The problem is that we would be depending on object equality, which is just identity. We would have nothing but cache misses, IIUC.